frontend/lib/src/components/elements/PlotlyChart/PlotlyChart.tsx

```diff
@@ -303,27 +299,58 @@ function PlotlyFigure({
     )
   }
 
-  const { data, layout, frames } = spec
-
   const reset = useCallback((): void => {
-    widgetMgr.setElementState(element.id, SELECTIONS, {})
-    widgetMgr.setElementState(element.id, DATA, {})
     widgetMgr.setStringValue(element, "{}", { fromUi: true }, fragmentId)
```

Comment:
I think we should check if the dragmode is `lasso` or `select` here as only the double click in that mode will reset the selection, even when a single selection is made:
```
const dragmode = plotlyFigure.layout.dragmode
    if ((dragmode && dragmode === "lasso") || dragmode == "select") {
      widgetMgr.setStringValue(element, "{}", { fromUi: true }, fragmentId)
    }
```

https://github.com/streamlit/streamlit/assets/103006371/5fad7c48-559d-4ab4-b373-b88155fc8af7